device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java

```diff
@@ -110,12 +114,16 @@ public void open() throws IllegalStateException, IOException
 
             try
             {
-                this.deviceIO.registerMultiplexedDeviceClient(configList);
+                this.deviceIO.registerMultiplexedDeviceClient(configList, DEFAULT_REGISTRATION_TIMEOUT_MILLISECONDS);
             }
             catch (InterruptedException e)
             {
                 throw new IOException("Interrupted while registering device clients to the multiplexed connection", e);
             }
+            catch (MultiplexingClientException e)
+            {
+                throw new IOException("Failed to register one or more device clients to the multiplexed connection", e);
```

Comment:
can you elaborate why you change the exception type here? I am sure you have documented it somewhere in this PR but I am having a hard time following :D

Reply:
probably because of the method signature "public void open() throws IllegalStateException, IOException"